test/controllers/AdditionalPaymentAmountControllerSpec.scala

```diff
@@ -48,80 +46,60 @@ class AdditionalPaymentAmountControllerSpec extends SpecBaseWithApplication with
   def getRequest(method: String, idx: Int) =
     FakeRequest(method, additionalPaymentAmountRoute(idx)).withCSRFToken.asInstanceOf[FakeRequest[AnyContentAsEmpty.type]]
 
+  val mockRepository = mock[SessionRepository]
```

Comment:
as you are refactoring this, why don't we use `scalamock` instead of mockito ? mockito just throws null pointer exceptions when call is not correctly mocked, which are not helpful

Reply:
sure, i think mockito was already there.....will do next PR

Reply:
sure, nop